From 73ad0a9cb6fb2c0a794ce77a92095159563c73ae Mon Sep 17 00:00:00 2001
From: Conor McCarthy <cmccarthy@codeweavers.com>
Date: Tue, 10 Dec 2024 00:23:16 +1000
Subject: [PATCH] mf: Do not set the topo status if a source has been shut
 down.

The status must remain MF_TOPOSTATUS_INVALID because the topology is
invalid and must be handled as such in session_start().

Fixes a regression introduced by efb369f5755067829df2aafe9b270d59cf5090ac.
---
 dlls/mf/session.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/mf/session.c b/dlls/mf/session.c
index fa6938ca193..5b711f7712d 100644
--- a/dlls/mf/session.c
+++ b/dlls/mf/session.c
@@ -516,7 +516,7 @@ static void session_set_topo_status(struct media_session *session, HRESULT statu
     IMFMediaEvent *event;
     PROPVARIANT param;
 
-    if (topo_status == MF_TOPOSTATUS_INVALID)
+    if (topo_status == MF_TOPOSTATUS_INVALID || status == MF_E_SHUTDOWN)
         return;
 
     if (list_empty(&session->topologies))
-- 
GitLab

